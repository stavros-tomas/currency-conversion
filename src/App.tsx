import './App.css'
import Convert from './components/Convert/Convert'
import { useEffect, useState } from 'react'
import type { CurrenciesResponse } from './types'

function App() {
  document.title = 'Currency Conversion';

  const API_KEY = import.meta.env.VITE_SECRET_CURRENCYBEACON_API_KEY;
  const CURRENCIES_BASE_API_URL = import.meta.env.VITE_CURRENCYBEACON_CURRENCIES_API_URL;

  const [currencies, setCurrencies] = useState<CurrenciesResponse[]>([]);

  const fetchCurrencies = async() => {
    if (!API_KEY || !CURRENCIES_BASE_API_URL) {
      console.error('API key or Base URL is missing');
      return;
    }

    const CURRENCIES_API_URL = `${CURRENCIES_BASE_API_URL}?api_key=${API_KEY}`;

    try {
      const response = await fetch(CURRENCIES_API_URL)
      const data = await response.json()

      setCurrencies(data.response);
    } catch (error) {
      console.error(error)
    }
  }

  useEffect(() => {
    fetchCurrencies()
  }, []);

  return (
    <div className="tw-flex tw-flex-col tw-items-center tw-p-4">
      <h1 className="tw-text-3xl tw-font-bold">Currency Conversion</h1>

      <div className="tw-mt-10 tw-w-full tw-max-w-xl">
        <Convert currencies={currencies} />
      </div>
    </div>
  )
}

export default App
