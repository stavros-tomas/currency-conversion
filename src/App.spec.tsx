import { render, screen } from '@testing-library/react';
import App from './App';
import { describe, it, expect } from 'vitest';

describe('App', () => {
  it('should render the title', () => {
    render(<App />);
    const titleElement = screen.getByText('Currency Conversion');
    expect(titleElement).toBeInTheDocument();
  });

  it('should render as expected', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
