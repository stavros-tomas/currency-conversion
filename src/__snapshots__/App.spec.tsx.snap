// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render as expected 1`] = `
<div>
  <div
    class="tw-flex tw-flex-col tw-items-center tw-p-4"
  >
    <h1
      class="tw-text-3xl tw-font-bold"
    >
      Currency Conversion
    </h1>
    <div
      class="tw-mt-10 tw-w-full tw-max-w-xl"
    >
      <div
        class="_convertWrapper_6abe82"
      >
        <div
          class="tw-mt-4 tw-w-full"
        >
          <div
            class="MuiFormControl-root tw-w-full css-1iw3t7y-MuiFormControl-root"
          >
            <div
              class="tw-flex tw-flex-row"
            >
              <div
                class="tw-w-4/12"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  id="From"
                >
                  From
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl tw-w-full css-sc8y68-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                >
                  <div
                    aria-controls=":r4:"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="From"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="combobox"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-81qg8w"
                    >
                      <span>
                        From
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="tw-w-8/12 tw-ml-2"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root tw-w-full css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="filled-basic"
                    id="filled-basic-label"
                  >
                    Amount
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id="filled-basic"
                      type="number"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-81qg8w"
                      >
                        <span>
                          Amount
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-mt-6 tw-w-full"
        >
          <div
            class="MuiFormControl-root tw-w-full css-1iw3t7y-MuiFormControl-root"
          >
            <div
              class="tw-flex tw-flex-row"
            >
              <div
                class="tw-w-4/12"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  id="To"
                >
                  To
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl tw-w-full css-sc8y68-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                >
                  <div
                    aria-controls=":r6:"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="To"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="combobox"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-81qg8w"
                    >
                      <span>
                        To
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="tw-w-8/12 tw-ml-2"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root tw-w-full css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="filled-basic"
                    id="filled-basic-label"
                  >
                    Amount
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id="filled-basic"
                      type="number"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-81qg8w"
                      >
                        <span>
                          Amount
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-w-full tw-mt-8"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary tw-w-full css-1abvrni-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Convert
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
